### DISCUSSION TITLE
Issues with windows build

### DISCUSSION THREAD

**Discussion by dbf256** [organicmaps/organicmaps#2101] [Q&A]
I understand that core dev team doesn't use windows, but I have two issuses, maybe you can advice.
1) For android build I have an error

```
CreateProcess error=193, %1 is not a valid Win32 application
A problem occurred starting process 'command '../tools/unix/version.sh''
```

Is it possible to have android studio running in Windows to use WSL for compilation?

2) For desktop app I try to build it using command-line in WSL and have cmake error. Any suggestions how to fix it?

```
Determining if the arc4random_buf exist failed with the following output:
Change Dir: /mnt/d/Projects/omim-build-release/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/ninja cmTC_4412f && [1/2] Building C object CMakeFiles/cmTC_4412f.dir/CheckSymbolExists.c.o
FAILED: CMakeFiles/cmTC_4412f.dir/CheckSymbolExists.c.o 
/usr/bin/cc   -fPIE -o CMakeFiles/cmTC_4412f.dir/CheckSymbolExists.c.o -c CheckSymbolExists.c
CheckSymbolExists.c: In function ‘main’:
CheckSymbolExists.c:8:19: error: ‘arc4random_buf’ undeclared (first use in this function)
    8 |   return ((int*)(&arc4random_buf))[argc];
      |                   ^~~~~~~~~~~~~~
CheckSymbolExists.c:8:19: note: each undeclared identifier is reported only once for each function it appears in
ninja: build stopped: subcommand failed.



```

Thanks.

**Comment by biodranik** [comment]
Regarding 1, if you have WSL, try to run `bash ../tools/version.sh` from the gradle file (or whatever full path to bash you have there).

Regarding 2, please share a full log. Did you run ./configure.sh? Are you compiling for Windows desktop?

**Comment by dbf256** [comment]
@biodranik for 2) I compile it from WSL Ubuntu, so should be a linux build.
./configure.sh is fine and full log for `tools/unix/build_omim.sh -r desktop` is attached.
THanks.
[CMakeOutput.log](https://github.com/organicmaps/organicmaps/files/8055342/CMakeOutput.log)

[CMakeError.log](https://github.com/organicmaps/organicmaps/files/8055341/CMakeError.log)

**Reply by biodranik** [reply]
We never compiled our code with WSL, looks like GCC in it lacks some implementations. Did you try with clang?

I would recommend you to try compile our code with Visual Studio compiler. 8 years ago I did it successfully, even without CMake :)

I know that CMake supports Visual Studio generator, probably some other code changes may be necessary to compile it for desktop properly. I can help you if some questions arise.
